clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -8007,10 +8007,29 @@ static bool checkSYCLAddIRAttributesMergeability(const AddIRAttrT &NewAttr,
 
 void Sema::CheckSYCLAddIRAttributesFunctionAttrConflicts(Decl *D) {
   const auto *AddIRFuncAttr = D->getAttr<SYCLAddIRAttributesFunctionAttr>();
-  if (!AddIRFuncAttr || AddIRFuncAttr->args_size() == 0 ||
+
+  // If there is no such attribute there is nothing to check. If there are
+  // dependent arguments we cannot know the actual number of arguments so we
+  // defer the check.
+  if (!AddIRFuncAttr ||
       hasDependentExpr(AddIRFuncAttr->args_begin(), AddIRFuncAttr->args_size()))
     return;
 
+  // If there are no name-value pairs in the attribute it will not have an
+  // effect and we can skip the check. The filter is ignored.
+  size_t NumArgsWithoutFilter =
+      AddIRFuncAttr->args_size() - (AddIRFuncAttr->hasFilterList() ? 1 : 0);
+  if (NumArgsWithoutFilter == 0)
+    return;
+
+  // "sycl-single-task" is present on all single_task invocations, implicitly
```

Comment:
Where is the conflict with `max_global_work_dim `checked?

Reply:
sycl/test/check_device_code/fpga_single_task_max_global_work_dim.cpp does some testing for this. We should not issue a warning here though as this behavior is implicit so overwriting it using the clang attribute should be allowed (if nothing else then to avoid breaking existing code.)

There could be an argument for leveraging this knowledge to check that `[[intel::max_global_work_dim]]` is `0` when applied to a `single_task`, but that seems outside the scope of this.